core/eth/reader.go

```diff
@@ -619,3 +640,9 @@ func (t *Reader) GetReservationWindow(ctx context.Context) (uint32, error) {
 	// contract is not implemented yet
 	return 0, nil
 }
+
+func (t *Reader) GetOperatorSocket(ctx context.Context, operatorId core.OperatorID) (string, error) {
```

Comment:
whenever we call this, should we check if the string is empty?

Reply:
🤔 I don't think the check is necessary, we will return whatever the contract returns with an empty string. Although this is a local optimization we can make, if we expect the input operator id is ever empty?

Reply:
hmm should we just place that check in this method? i.e. return error if result is empty

Reply:
If the result is empty, then the check against the expected operator socket would fail. Is there a reason we want to handle it here?

Reply:
updated to check for empty string here